src/libhirte/common/time-util.c

```diff
@@ -20,3 +22,25 @@ uint64_t finalize_time_interval_micros(int64_t start_time_micros) {
 double micros_to_millis(uint64_t time_micros) {
         return (double) time_micros * microsec_to_millisec_multiplier;
 }
+
+int get_formatted_timestamp(char *timebuf_ret, const size_t buf_size) {
+        struct timespec now;
```

Comment:
It would be great to have unit tests for this function (similar to [string-util_test.c](https://github.com/containers/hirte/blob/main/src/libhirte/test/common/string-util_test.c))

Reply:
Quite challenging to test it, as it returns current time, and it may fail from time to time if I get current time separately... Any ideas?

Reply:
You would need to pass current time using a parameter, that way you can set any time for testing purposes

Reply:
Yes, that would be the way to go. Sorry the (too) late reply.

Reply:
Done